isis/src/control/apps/findfeatures/GenericTiledTransform.cpp

```diff
@@ -0,0 +1,180 @@
+		int srcMaxY = (int)ceil(std::max(std::max(srcCorner00.y, srcCorner01.y), std::max(srcCorner10.y, srcCorner11.y)));
+		cv::Range srcRangeX(srcMinX, srcMaxX);
+		cv::Range srcRangeY(srcMinY, srcMaxY);
+		return std::make_pair(srcRangeX, srcRangeY);
+}
+
+std::pair<cv::Range, cv::Range> GenericTiledTransform::addSourceInterpMargin(const std::pair<cv::Range, cv::Range>& srcRangeXY,
+                                                                             const cv::Mat &image,
+                                                                             const int margin) const {
+	const cv::Range& srcRangeX = srcRangeXY.first;
+	const cv::Range& srcRangeY = srcRangeXY.second;
+	int srcMinX = std::max(srcRangeX.start - margin, 0);
+	int srcMaxX = std::min(srcRangeX.end   + margin, GenericTransform::getSize(image).width);
+	int srcMinY = std::max(srcRangeY.start - margin, 0);
+	int srcMaxY = std::min(srcRangeY.end   + margin, GenericTransform::getSize(image).height);
+	return std::make_pair(cv::Range(srcMinX, srcMaxX), cv::Range(srcMinY, srcMaxY));
+}
+
+cv::Mat_<cv::Vec2f> GenericTiledTransform::computeMapping(const GenericTiledTransform::RangeXY& srcRangeXY,
+                                                          const GenericTiledTransform::RangeXY& dstRangeXY) const {
+  const cv::Range& dstRangeX = dstRangeXY.first;
+  const cv::Range& dstRangeY = dstRangeXY.second;
+  const cv::Range& srcRangeX = srcRangeXY.first;
+  const cv::Range& srcRangeY = srcRangeXY.second;
+  cv::Mat_<cv::Vec2f> mapXY(cv::Size(dstRangeX.size(), dstRangeY.size()));
+  const cv::Point2f sourceTileOffset = cv::Point2i(srcRangeX.start, srcRangeY.start);
+  for (int dstY = dstRangeY.start; dstY < dstRangeY.end; ++dstY)
+  {
+    for (int dstX = dstRangeX.start; dstX < dstRangeX.end; ++dstX)
+    {
+      cv::Point2f dstXY = cv::Point2i(dstX, dstY);
+      cv::Point2f srcXY = inverse(dstXY) - sourceTileOffset;
+      mapXY(dstY - dstRangeY.start, dstX - dstRangeX.start) = srcXY;
+    }
+  }
+  return mapXY;
+}
+
+cv::Mat GenericTiledTransform::render(const cv::Mat &image) const{
+  cv::Size tFormSize = transformedSize(getMatrix(), image.size()).size();
+  cv::Mat result = cv::Mat(tFormSize, image.type());
+  std::tuple<int, int> xyTiles = getNumTiles(tFormSize);
+  int nxTiles = std::get<0>(xyTiles);
+  int nyTiles = std::get<1>(xyTiles);
+  int n_tiles = nxTiles*nyTiles;
+  for (int i = 0; i < n_tiles; i++){
+    const std::pair<cv::Range, cv::Range> dstRangeXY = getTile(i, tFormSize);
+    const std::pair<cv::Range, cv::Range> srcRangeXY = addSourceInterpMargin(computeSourceRange(dstRangeXY), image, 3);
+    const cv::Range& dstRangeX = dstRangeXY.first;
+		const cv::Range& dstRangeY = dstRangeXY.second;
+		const cv::Range& srcRangeX = srcRangeXY.first;
+		const cv::Range& srcRangeY = srcRangeXY.second;
+		const cv::Mat_<cv::Vec2f> mapXY = computeMapping(srcRangeXY, dstRangeXY);
```

Comment:
What's with the weird indents here?

Reply:
The indent is only showing up for me on git. it looks perfectly fine in my editor.  I'm not sure how to fix it since I can't see it locally

Reply:
I downloaded the file locally and these lines are indented with tab characters. Check your tab -> space conversion. ISIS only uses spaces in the source code.